awx/main/models/schedules.py

```diff
@@ -248,11 +256,39 @@ def update_computed_fields(self):
                 self.dtend = future_rs[-1].astimezone(pytz.utc)
             except IndexError:
                 self.dtend = None
+
+        changed = any(getattr(self, field_name) != starting_values[field_name] for field_name in affects_fields)
+        return changed
+
+    def update_computed_fields(self):
+        changed = self.update_computed_fields_no_save()
+        if not changed:
+            return
         emit_channel_notification('schedules-changed', dict(id=self.id, group_name='schedules'))
+        # Must save self here before calling unified_job_template computed fields
+        # in order for that method to be correct
+        self.rrule = Schedule.coerce_naive_until(self.rrule)
```

Comment:
This line would have no effect, because it's not in the update_fields. It was a carry-over item from `.save`, and after reflecting on this, I think it should be removed because the method `coerce_naive_until` should have no dependency on the _current_ time, and will be deterministic given the input rrule. If you are here, the coercion has already been done on create / save.